clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/BlobSnapshotManager.java

```diff
@@ -150,6 +149,17 @@ public BlobTransferPartitionMetadata getTransferMetadata(BlobTransferPayload pay
    * @param blobTransferRequest the blob transfer request
    */
   private void recreateSnapshotAndMetadata(BlobTransferPayload blobTransferRequest) {
+    // Only create a new snapshot if there is no concurrent user.
+    // Otherwise, the snapshot is still in use and being transferred, and should not be recreated.
+    if (getConcurrentSnapshotUsers(blobTransferRequest.getTopicName(), blobTransferRequest.getPartition()) != 0L) {
```

Comment:
I am wondering if we also make stale snapshot threshold configurable? I think for now it is 30mins by default, but for certain heavy hybrid use case we might want to lower the threshold down on the app side.

Reply:
snapshot retention time is configurable with default value 60, `snapshotRetentionTimeInMin = serverProperties.getInt(BLOB_TRANSFER_SNAPSHOT_RETENTION_TIME_IN_MIN, 60);`